cmd/rule-evaluator/main.go

```diff
@@ -96,155 +106,107 @@ func main() {
 	// we reuse that constant.
 	version, err := export.Version()
 	if err != nil {
-		//nolint:errcheck
-		level.Error(logger).Log("msg", "Unable to fetch module version", "err", err)
+		_ = level.Error(logger).Log("msg", "Unable to fetch module version", "err", err)
 		os.Exit(1)
 	}
-	newExporter := exportsetup.FromFlags(a, fmt.Sprintf("rule-evaluator/%s", version))
-
-	notifierOptions := notifier.Options{Registerer: reg}
 
-	projectID := a.Flag("query.project-id", "Project ID of the Google Cloud Monitoring scoping project to evaluate rules against.").
-		Default(defaultProjectID).String()
-
-	targetURL := a.Flag("query.target-url", fmt.Sprintf("The address of the Prometheus server query endpoint. (%s is replaced with the --query.project-id flag.)", projectIDVar)).
-		Default(fmt.Sprintf("https://monitoring.googleapis.com/v1/projects/%s/location/global/prometheus", projectIDVar)).
-		String()
-
-	generatorURLStr := a.Flag("query.generator-url", "The base URL used for the generator URL in the alert notification payload. Should point to an instance of a query frontend that accesses the same data as --query.target-url.").
-		PlaceHolder("<URL>").String()
-
-	queryCredentialsFile := a.Flag("query.credentials-file", "Credentials file for OAuth2 authentication with --query.target-url.").
-		Default("").String()
-
-	disableAuth := a.Flag("query.debug.disable-auth", "Disable authentication (for debugging purposes).").
-		Default("false").Bool()
+	exporterOpts := export.ExporterOpts{
+		UserAgentProduct: fmt.Sprintf("rule-evaluator/%s", version),
+	}
+	exportsetup.ExporterOptsFlags(a, &exporterOpts)
```

Comment:
Hm, that changes the flag surface completely, break the users essentially, right? (`export.` prefix on all flags https://github.com/GoogleCloudPlatform/prometheus-engine/blob/afe2103347608bf63eb7555d208d638465dd3d7f/pkg/export/setup/setup.go)

Do we want this?

Reply:
Please see https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1001 first. On the rule-evaluator, we always had two sets of flags, query and export. Those have not changed here, but split up and exposed in the public API. The query options are now part of the `evaluatorOptions`.

Reply:
PR: https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1053

Reply:
Done. Let me know what you think of it!